src/signtool/IBM_Container.cpp

```diff
@@ -0,0 +1,1767 @@
+/* IBM_PROLOG_BEGIN_TAG                                                   */
+/* This is an automatically generated prolog.                             */
+/*                                                                        */
+/* $Source: src/signtool/IBM_Container.cpp $                              */
+/*                                                                        */
+/* OpenPOWER sb-signing-utils Project                                     */
+/*                                                                        */
+/* Contributors Listed Below - COPYRIGHT 2016                             */
+/* [+] International Business Machines Corp.                              */
+/*                                                                        */
+/*                                                                        */
+/* Licensed under the Apache License, Version 2.0 (the "License");        */
+/* you may not use this file except in compliance with the License.       */
+/* You may obtain a copy of the License at                                */
+/*                                                                        */
+/*     http://www.apache.org/licenses/LICENSE-2.0                         */
+/*                                                                        */
+/* Unless required by applicable law or agreed to in writing, software    */
+/* distributed under the License is distributed on an "AS IS" BASIS,      */
+/* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or        */
+/* implied. See the License for the specific language governing           */
+/* permissions and limitations under the License.                         */
+/*                                                                        */
+/* IBM_PROLOG_END_TAG                                                     */
+
+#include <string.h>
+#include <stddef.h>
+
+#include <endian.h>
+
+#include <sstream>
+#include <fstream>
+#include <iostream>
+#include <algorithm>
+
+#include "IBM_Crypto.h"
+#include "IBM_Exception.h"
+#include "IBM_HexString.h"
+#include "IBM_Container.h"
+#include "IBM_Utils.h"
+
+namespace
+{
+    const static uint32_t g_COL_WIDTH = 24;
+
+    const static uint32_t g_UNCOMPRESSED_PUBKEY_FORMAT = 0x04;
+
+    const static std::string g_HASH_ALGO_SHA512_NAME   = "SHA-512";
+    const static std::string g_SIGN_ALGO_ECDSA521_NAME = "ECDSA521";
+
+    template<typename T_PAIR>
+    struct GetKey: public std::unary_function<T_PAIR, typename T_PAIR::first_type>
+    {
+        const typename T_PAIR::first_type& operator()(const T_PAIR& item) const
+        {
+            return item.first;
+        }
+    };
+
+    uint16_t getUint16( const uint8_t *data )
```

Comment:
get it from what? In what format?

Why are you writing custom endian routines? What exactly is wrong with hbe16toh(3) in endian.h ?